elm/web/rhub.py

```diff
@@ -585,22 +585,25 @@ def authors(self):
         """
         pa = self.get('personAssociations')
 
+        if not pa:
+            return None
+
         authors = []
 
         for r in pa:
-            first = r.get('name').get('firstName')
-            last = r.get('name').get('lastName')
+            name = r.get('name')
+
+            if not name:
+                continue
 
-            if first and last:
-                full = first + ' ' + last
-            elif first:
-                full = first
-            elif last:
-                full = last
+            first = name.get('firstName')
+            last = name.get('lastName')
+
+            full = " ".join(filter(bool, [first, last]))
```

Comment:
Interestingly, writing it like this illuminates a case that  was overlooked in the original implementation: what if both `first` and `last` are `None`? Is that even possible? Is this something you want to handle (e.g.  something like `if not full: full = "Unknown author"`, or maybe just never add it to the `authors` list)?

I'll leave it up to your discretion

Reply:
At the moment every entry in our db has a value for authors and I am pretty sure it's a requirement for the research hub api, regardless I will add some logic that excludes it from the list just in case.